packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -215,9 +215,16 @@ class OverlayState extends State<Overlay> {
   }
 
   void _remove(OverlayEntry entry) {
-    setState(() {
-      _entries.remove(entry);
+    assert(() {
+      if (!mounted)
+        assert(_entries.remove(entry));
```

Comment:
It's n^2.  Maybe we should make _entries be a LinkedHashSet rather than a List so that remove is constant time.